### PR TITLE
set kruize local to true 

### PR DESCRIPTION
**Pull request by msvinaykumar**
## Description

make local=true by default

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [ ] New Test X
- [ ] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: 

## Checklist :dart:

- [ ] Followed coding guidelines
- [ ] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

Include any additional information such as links, test results, screenshots here

### CODE REVIEW

**Comment by dinogun** [comment]
PR checks are failing, I guess tests need to be updated?

**Comment by msvinaykumar** [comment]
@chandrams is there a way to switch test suite from remote_monitoring_tests to local_monitoring_tests
so that it uses 
            /createMetricProfile

**Comment by chandrams** [comment]
@msvinaykumar - As discussed, all remote monitoring tests will fail if local flag is set to true. To resolve this, we need to set the local flag to false for all remote monitoring tests.  And in local monitoring tests, you can remove the code to set it to true as the default will be set to true.

**Comment by msvinaykumar** [comment]
Thanks @chandrams 
im seeing this issue now

`     @pytest.mark.test_e2e
    def test_list_recommendations_multiple_exps_for_datasource_workloads(cluster_type):
        """
        Test Description: This test validates list recommendations for multiple experiments posted using different json files
        """
        clone_repo("https://github.com/kruize/benchmarks")
        benchmarks_install()
        container_id = apply_tfb_load("default", cluster_type)
        print(container_id)
    
        # list all datasources
        form_kruize_url(cluster_type)
    
        # Get the datasources name
        datasource_name = None
        response = list_datasources(datasource_name)
    
        list_datasources_json = response.json()
    
        assert response.status_code == SUCCESS_200_STATUS_CODE
    
        # Validate the json against the json schema
        errorMsg = validate_list_datasources_json(list_datasources_json, list_datasources_json_schema)
>       assert errorMsg == ""
E       assert [<ValidationError: "'datasources' is a required property">, <ValidationError: ''>] == '`

**Comment by chandrams** [comment]
@msvinaykumar - From the kruize pod log in results, I see datasource is not servicable

```
2024-10-1506:21:41.532 INFO [main][CloudWatchAppender.java(118)]-AWS access details are not provided. Skipping sending logs to CloudWatch.
2024-10-1506:21:41.924 INFO [main][KruizeHibernateUtil.java(64)]-DB is trying to connect to jdbc:postgresql://kruize-db-service:5432/kruizeDB
2024-10-1506:21:43.355 INFO [main][KruizeHibernateUtil.java(66)]-DB build session is successful !
2024-10-1506:21:43.490 INFO [main][Autotune.java(264)]-DB creation successful !
2024-10-1506:21:43.492 INFO [main][Autotune.java(272)]-DB Liveliness probe connection successful!
2024-10-1506:21:43.493 INFO [main][Autotune.java(110)]-Now running kruize local DDL's 
2024-10-1506:21:43.513 INFO [main][Autotune.java(264)]-DB creation successful !
2024-10-1506:21:43.513 INFO [main][Autotune.java(272)]-DB Liveliness probe connection successful!
2024-10-1506:21:43.515 INFO [main][DataSourceCollection.java(48)]-Checking available datasources from database:
2024-10-1506:21:43.726 INFO [main][DataSourceCollection.java(51)]-No datasource found in database.
2024-10-1506:21:43.780 INFO [main][DataSourceCollection.java(89)]-Trying to add the datasource to collection: prometheus-1
2024-10-1506:21:43.781 INFO [main][DataSourceCollection.java(97)]-Verifying datasource reachability status: prometheus-1
2024-10-1506:21:43.995 INFO [main][GenericRestApiClient.java(97)]-Executing request: GET http://prometheus-k8s.monitoring.svc.cluster.local:9090/api/v1/query?query=up HTTP/1.1
2024-10-1506:21:44.034 ERROR [main][DataSourceCollection.java(119)]-Datasource is not serviceable.
2024-10-1506:21:44.035 INFO [main][Autotune.java(191)]-Checking available datasources:
2024-10-1506:21:44.035 INFO [main][MetricProfileCollection.java(45)]-Checking available metric profiles from database: 
2024-10-1506:21:44.108 INFO [main][MetricProfileCollection.java(49)]-No metric profile found in database.
2024-10-1506:21:44.125 INFO [main][KruizeHibernateUtil.java(64)]-DB is trying to connect to jdbc:postgresql://kruize-db-service:5432/kruizeDB
2024-10-1506:21:44.358 INFO [main][KruizeHibernateUtil.java(66)]-DB build session is successful !
2024-10-1506:21:44.814 WARN [main][KubernetesServicesImpl.java(425)]-Watcher not added! Only REST API access is enabled.
2024-10-1506:21:44.923 WARN [main][KubernetesServicesImpl.java(425)]-Watcher not added! Only REST API access is enabled.
2024-10-1506:21:44.925 WARN [main][KubernetesServicesImpl.java(425)]-Watcher not added! Only REST API access is enabled.
2024-10-1506:21:44.926 INFO [main][ExperimentManager.java(32)]-EM version: testv1
2024-10-1506:21:44.927 INFO [main][ExperimentManager.java(28)]-Initializing EM
2024-10-1506:21:44.927 INFO [main][ExperimentManager.java(71)]-Adding EM Servlets
2024-10-1506:21:44.928 INFO [main][EMExecutorService.java(30)]-Creating regular executor
2024-10-1506:21:44.929 INFO [main][EMExecutorService.java(34)]-Creating scheduled executor
2024-10-1506:21:44.929 INFO [main][EMExecutorService.java(67)]-Starting stage processors
2024-10-1506:21:44.929 INFO [main][EMExecutorService.java(67)]-Starting stage processors
Waiting for experiments ... 
2024-10-1506:23:15.126 INFO [qtp1100399456-52][MetricProfileCollection.java(66)]-Trying to add the metric profile to collection: resource-optimization-local-monitoring
2024-10-1506:23:15.126 INFO [qtp1100399456-52][MetricProfileCollection.java(71)]-MetricProfile added to the collection successfully: resource-optimization-local-monitoring
2024-10-1506:23:44.076 ERROR [qtp1100399456-46][ListDatasourcesAPIObject.java(41)]-No datasources found
2024-10-1506:23:52.023 ERROR [qtp1100399456-50][ListDatasourcesAPIObject.java(41)]-No datasources found
```

File: tests/scripts/common/common_functions.sh [hunk 1]
```diff
@@ -1902,6 +1902,22 @@ function kruize_local_patch() {
...
+# "local" flag is turned off for RM.
+#
+function kruize_remote_patch() {
+	CRC_DIR="./manifests/crc/default-db-included-installation"
+	KRUIZE_CRC_DEPLOY_MANIFEST_OPENSHIFT="${CRC_DIR}/openshift/kruize-crc-openshift.yaml"
+	KRUIZE_CRC_DEPLOY_MANIFEST_MINIKUBE="${CRC_DIR}/minikube/kruize-crc-minikube.yaml"
+
+
+  if [ ${cluster_type} == "minikube" ]; then
+    sed -i 's/"local": "false"/"local": "false"/' ${KRUIZE_CRC_DEPLOY_MANIFEST_MINIKUBE}
```
[the remark this answers is not in the document]
Comment (msvinaykumar) [reply tests/scripts/common/common_functions.sh:1915]: ok

**Comment by msvinaykumar** [comment]
> @msvinaykumar Please update other non-functional test scripts too

done

File: tests/scripts/remote_monitoring_tests/stress_test/remote_monitoring_stress_test.sh [hunk 2]
```diff
@@ -130,6 +130,8 @@ jmeter_setup
 echo "Invoking jmeter setup...done" | tee -a ${LOG}
 
 echo "Setting up kruize..." | tee -a ${LOG}
+echo "setting local=false"
+kruize_remote_patch
```
Comment (chandrams) [inline tests/scripts/remote_monitoring_tests/stress_test/remote_monitoring_stress_test.sh:134]: This should be moved inside the push

**Review by dinogun — APPROVED** [review]
Merging this pending remaining remote_monitoring tests getting fixed in a separate PR

**Pull request merged**